site/pages/api/publish-edit-template.api.ts

```diff
@@ -0,0 +1,122 @@
+            return;
+        }
+
+        const [draftDisruption, orgInfo] = await Promise.all([
+            getDisruptionById(validatedBody.data.disruptionId, session.orgId, true),
+            getOrganisationInfoById(session.orgId),
+        ]);
+
+        if (!orgInfo) {
+            logger.error(`Organisation info not found for Org Id ${session.orgId}`);
+            redirectTo(res, ERROR_PATH);
+            return;
+        }
+        if (!draftDisruption || Object.keys(draftDisruption).length === 0) {
+            logger.error(`Disruption ${validatedBody.data.disruptionId} not found to publish`);
+            redirectTo(res, ERROR_PATH);
+
+            return;
+        }
+
+        const validatedDisruptionBody = publishDisruptionSchema.safeParse(draftDisruption);
+
+        if (!validatedDisruptionBody.success) {
+            setCookieOnResponseObject(
+                COOKIES_TEMPLATE_OVERVIEW_ERRORS,
+                JSON.stringify(flattenZodErrors(validatedDisruptionBody.error)),
+                res,
+            );
+
+            redirectTo(res, `${TEMPLATE_OVERVIEW_PAGE_PATH}/${validatedBody.data.disruptionId}`);
+            return;
+        }
+
+        const isEditPendingDsp =
+            draftDisruption.publishStatus === PublishStatus.pendingAndEditing ||
+            draftDisruption.publishStatus === PublishStatus.editPendingApproval;
+
+        if (isEditPendingDsp) {
+            await publishEditedConsequencesAndSocialMediaPostsIntoPending(
+                draftDisruption.disruptionId,
+                session.orgId,
+                true,
+            );
+        } else {
+            await publishEditedConsequencesAndSocialMediaPosts(draftDisruption.disruptionId, session.orgId, true);
+        }
+
+        if (canPublish(session)) {
+            if (isEditPendingDsp)
+                await publishPendingConsequencesAndSocialMediaPosts(draftDisruption.disruptionId, session.orgId, true);
+            await Promise.all([
+                deleteDisruptionsInEdit(draftDisruption.disruptionId, session.orgId, true),
+                deleteDisruptionsInPending(draftDisruption.disruptionId, session.orgId, true),
+            ]);
+        } else {
+            await deleteDisruptionsInEdit(draftDisruption.disruptionId, session.orgId, true);
+        }
+
+        draftDisruption.publishStatus === PublishStatus.pendingAndEditing && !canPublish(session)
+            ? await updatePendingDisruptionStatus(
+                  { ...draftDisruption, publishStatus: PublishStatus.editPendingApproval },
+                  session.orgId,
+                  true,
+              )
+            : await insertPublishedDisruptionIntoDynamoAndUpdateDraft(
+                  draftDisruption,
+                  session.orgId,
+                  canPublish(session) ? PublishStatus.published : PublishStatus.pendingApproval,
+                  session.name,
+                  undefined,
+                  true,
+              );
```

Comment:
Can this logic be removed/ simplified if we do not require approvals for templates?